EIPS/eip-5792.md

```diff
@@ -122,38 +136,55 @@ Returns the status of a call batch that was sent via `wallet_sendCalls`. The ide
 type GetCallsParams = string;
 
 type GetCallsResult = {
-  status: 'PENDING' | 'CONFIRMED';
+  batchId: `0x${string}`;
+  chainId?: `0x${string}`;
+  batchStatus: string; // See "Status Codes"
   receipts?: {
     logs: {
       address: `0x${string}`;
       data: `0x${string}`;
       topics: `0x${string}`[];
     }[];
     status: `0x${string}`; // Hex 1 or 0 for success or failure, respectively
-    chainId: `0x${string}`;
     blockHash: `0x${string}`;
     blockNumber: `0x${string}`;
     gasUsed: `0x${string}`;
     transactionHash: `0x${string}`;
   }[];
+  capabilitiesResultsData?: Record<string, any> | undefined;
 };
 ```
 
+##### Status Codes for `batchStatus` field
+
+The purpose of the `batchStatus` field is to provide a short summary of the current status of the batch.
+It provides some off-chain context to the array of inner transaction `receipts`.
+
+| Name      | Description                                                                                                          |
+|-----------|----------------------------------------------------------------------------------------------------------------------|
+| PENDING   | Batch has been received by the wallet but has not completed execution on-chain                                       |
```

Comment:
How do you feel about possibly expanding `PENDING` to capture more of the offchain context around the pending state?

For example

- `PENDING`: Batch has been received by the wallet but has not completed execution on-chain 
- `UNDERPRICED`: Batch has been received by the wallet but the gas prices are too low for the current chain conditions
- `STALLED`: Batch has been received by the wallet but is stuck for non-pricing related reasons; i.e. ERC-4337 bundler is down. The definition of 'stuck' is left up to the wallet, i.e. batch is priced appropriately but has not been included in >60s

Reply:
@forshtat would be good to get your thoughts here. imo `UNDERPRICED` specifically seems like a reasonable addition, as it could be retried.

@ajhodges i'm not a huge fan of the ambiguity for `STALLED` though. to me this is effectively still `PENDING`, and each wallet should have their own interpretations of `PENDING` taking "too long". if we and others feel strongly about it being its own status, i would prefer to be more opinionated in the spec and define when a `PENDING` batch becomes `STALLED`, instead of leaving it open-ended.

Reply:
What if we just left it as `PENDING` and `SUCCESS`, and had a sibling `statusCode` or `detail` property that could be extensible enough to accommodate for what ever addition? 
This would make it a lot easier to coerce on frontend apps.

Reply:
> `UNDERPRICED` specifically seems like a reasonable addition, as it could be retried.

I don't have any strong feelings against `UNDERPRICED` or any other status code we can clearly define.

About `UNDERPRICED` specifically, I guess its difference with the regular `PENDING` state is that the user needs to confirm a modification to the original request? And if the wallet could just increase the price internally and retry, it would not be different from `PENDING`, would it?

If so, maybe a better broader category would be `ACTION_REQUIRED` or something like that, what do you think?

Reply:
Yeah, I guess `statusCode` might be more realistic than `detail` as an application can choose which message they want to display to an end-user (if anything). 

Perhaps this could be a similar pattern to HTTP status codes where `2xx` ≈ success, `4xx` ≈ client error, `5xx` ≈ server error, etc. For this case, maybe `1xx` represents `PENDING` sub-statuses, `2xx` represents `CONFIRMED` sub-statuses, and so on. For example, a `status` of `PENDING` and a `statusCode` of `101` could mean “stalled”, and an application could either choose to show a generic loading indicator, or a loading indicator with “this tx is taking a while, pls wait”. 

I guess the motivation of a sub-status field like `statusCode` is that it would make the branching logic within an application so much easier since 5792 is an ERC for applications.

Reply:
>I guess the motivation of a sub-status field like statusCode is that it would make the branching logic within an application so much easier — they can rely on the simplicity of status and choose to show extra information by utilizing statusCode.

Yep, exactly. As an app, I want to know if things are healthy. Is my transaction stuck? Is there anything I can do about it if it's stuck? (i.e. underpriced vs stalled). I think the display logic/messaging to the user should be left up to the app, but the wallet can oftentimes provide additional context about what happened to make the UX a little better.

I agree `UNDERPRICED` is maybe a bit vague. But the wallet should be the expert on the current chain conditions and the bundle pricing so it's best suited to detect if the calls should be retried (and we shouldn't prescribe any conditions in this EIP). So maybe `RETRYABLE` is a better sub-status than `UNDERPRICED`, signifying that the calls might go through quicker if you retry. I think I like this better than `ACTION_REQUIRED` because technically an underpriced tx can still go through if chain conditions change